### PR TITLE
feat: time restrictions start page

### PR DESCRIPTION
**Pull request by RobCatton**
# Description

-   To add a page which lets the user decide if there are time restrictions on their ticket or not.

# Testing instructions

-   Run site through dev repo and navigate through.

# Type of change

-   [x] feat - A new feature
-   [ ] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

# Checklist:

-   [x] Able to run pr locally
-   [x] Lighthouse performed (Accessibility 90+ minimum)
-   [x] Followed acceptance criteria
-   [x] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my own code
-   [x] I have made corresponding changes to the documentation if applicable
-   [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/pages/timeRestrictions.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                    </div>
+                    <input type="submit" value="Continue" id="continue-button" className="govuk-button" />
+                </>
+            </CsrfForm>
+        </TwoThirdsLayout>
+    );
+};
+
+export const getServerSideProps = (ctx: NextPageContextWithSession): { props: TimeRestrictionsProps } => {
+    const timeRestrictionsInfo = getSessionAttribute(ctx.req, TIME_RESTRICTIONS_ATTRIBUTE);
```
Comment (dannydaviesIW) [inline src/pages/timeRestrictions.tsx:101]: By removing the `TimeRestrictionsAttribute` from the API, this variable would then only ever be undefined or `TimeRestrictionsAttributeWithErrors`. Hence, we could have a very simple getServerSideProps which just sets an errors array 😄

File: tests/pages/api/passengerType.test.ts [hunk 2]
```diff
@@ -62,11 +61,12 @@ describe('passengerType', () => {
...
             mockWriteHeadFn: writeHeadMock,
         });
-        const redirectOnFareType = jest.spyOn(utils, 'redirectOnFareType');
 
         passengerType(req, res);
 
-        expect(redirectOnFareType).toHaveBeenCalled();
+        expect(writeHeadMock).toBeCalledWith(302, {
+            Location: '/timeRestrictions',
+        });
```
Comment (dannydaviesIW) [inline tests/pages/api/passengerType.test.ts:69]: Looks like this test needs its name adjusting a little

File: tests/pages/api/timeRestrictions.test.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import { TIME_RESTRICTIONS_ATTRIBUTE } from '../../../src/constants';
+
+beforeEach(() => {
+    (isSessionValid as {}) = jest.fn().mockReturnValue(true);
+});
+
+afterEach(() => {
+    jest.resetAllMocks();
+});
+
```
Comment (dannydaviesIW) [inline tests/pages/api/timeRestrictions.test.ts:16]: Should these `beforeEach` and `afterEach` functions be inside the first 'describe'?

File: tests/pages/api/timeRestrictions.test.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        const { req, res } = getMockRequestAndResponse({
+            cookieValues: {},
+            body: {},
+            uuid: {},
+        });
+        timeRestrictions(req, res);
+        expect(res.writeHead).toBeCalledWith(302, {
+            Location: '/timeRestrictions',
+        });
+        console.log(updateSessionAttributeSpy.mock.calls[0][2]);
```
Comment (dannydaviesIW) [inline tests/pages/api/timeRestrictions.test.ts:41]: Sneaky console.log

File: tests/pages/timeRestrictions.test.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import TimeRestrictions from '../../src/pages/timeRestrictions';
+
+describe('pages', () => {
+    describe('timeRestrictions', () => {
+        it('should render correctly', () => {
+            const tree = shallow(
+                <TimeRestrictions timeRestrictionsInfo={{ timeRestrictions: false }} csrfToken="" pageProps={[]} />,
+            );
+            expect(tree).toMatchSnapshot();
+        });
```
Comment (dannydaviesIW) [inline tests/pages/timeRestrictions.test.tsx:12]: We could have an additional snapshot test to check that the page correctly renders errors also

File: tests/pages/api/timeRestrictions.test.ts (hunk 3 above)
Comment (RobCatton) [reply tests/pages/api/timeRestrictions.test.ts:16]: As theres only one describe, works either way.

**Pull request merged**